pkg/database/db.go

```diff
@@ -7,26 +7,37 @@
 package database
 
 import (
+	"bufio"
 	"bytes"
+	"compress/zlib"
+	"encoding/binary"
 	"encoding/gob"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"io"
 	"os"
+	"path"
 	"path/filepath"
 	"sync"
 	"time"
 
 	"github.com/rs/zerolog"
 )
 
-type Database struct {
-	Version  int
-	Name     string
-	Path     string
-	Segments []Segment
-	Current  int
+// FossilDBVersion is the version of the database as recorded on disk.
+// This is primarily used for migration.
+const FossilDBVersion = 2
 
+type Database struct {
+	Version     uint32
+	Segments    []Segment
+	Current     uint32
 	TopicLookup []string
 	TopicCount  int
+	STime       time.Time // Last serialize time
```

Comment:
We should add this to stats. It would be a good metric to correlate with other metrics.